pymc/sampling.py

```diff
@@ -1687,6 +1708,20 @@ def compile_forward_sampling_function(
     if basic_rvs is None:
         basic_rvs = []
 
+    if constant_data is None:
+        constant_data = {}
+    if constant_coords is None:
+        constant_coords = set()
+
+    # We define a helper function to check if shared values match to an array
+    def shared_value_matches(var):
+        shared_value = var.get_value()
+        try:
+            array = constant_data[var.name]
+            return shared_value.shape == array.shape and all(shared_value == array)
```

Comment:
Does this do the same?
```suggestion
            return np.array_equal(shared_value, array)
```

Reply:
Looks like it. I had no idea it existed!